controllers/terminal_controller.go

```diff
@@ -1073,6 +1065,14 @@ func (r *TerminalReconciler) createOrUpdateTerminalPod(ctx context.Context, cs *
 			if !tolerationExists(pod.Spec.Tolerations, match(noExecuteToleration)) {
 				pod.Spec.Tolerations = append(pod.Spec.Tolerations, noExecuteToleration)
 			}
+
+			noScheduleToleration := corev1.Toleration{
```

Comment:
how about having only one toleration? Similar like it was done in the ops-toolbelt https://github.com/gardener/ops-toolbelt/pull/79/files

```
tolerations:
  - operator: Exists
```